tests/__init.test.js

```diff
@@ -77,6 +77,8 @@ teardown(function (done) {
   this.sinon.restore();
   delete AFRAME.components.test;
   delete AFRAME.systems.test;
+  // Reset global legacyMode flag in THREE
+  AFRAME.THREE.ColorManagement.legacyMode = true;
```

Comment:
Why is this needed for unit tests?

Reply:
Note that `legacyMode = true` is the (current) default value in THREE.js. (Re)setting it here ensures all unit tests start with the expected default value. Otherwise as soon as one unit test changes it to `false` (i.e. uses colorManagement), subsequent unit tests will run with the non-legacy behaviour of `THREE.Color`.

For most tests it doesn't actually matter, but for any test that works with and inspects the values of `THREE.Color` instances directly, it will cause differences. Setting it for all tests is the most effective.

Reply:
FYI .legacyMode=false renamed to .enabled=true in r150
https://github.com/mrdoob/three.js/blob/7b04b508e32275b18f5def78c40ca38f426b7b2a/src/math/ColorManagement.js#L80-L86
In r152 enabled=true is the default and some other renaming was done, see https://twitter.com/donrmccurdy/status/1648412730691297280

I guess this PR will be merged for aframe 1.5.0 with three r152 or later because this will break some third parties components. It will surely break my code, I have some `THREE.Color(cssColor).convertSRGBToLinear()` in some of my components that I'll need to remove.

Reply:
It makes sense to merge it for 1.5.0 and to also switch the default to being color managed, matching Three.js. There will indeed be slight differences in behaviour, which I'm afraid is unavoidable. In a good chunk of cases this will likely not be an issue (color management can be disabled matching old defaults) and in some cases will actually fix broken behaviour ("old" components not converting to linear). I hadn't considered the manual usage of `.convertSRGBToLinear()` as I tend to use `rendererSystem.applyColorCorrection(color)`, but that would indeed break.

To summarize:
|                                | **Pre 1.5.0 (colorManagement: false)** | **Pre 1.5.0 (colorManagement: true)** | **1.5.0 (colorManagement: false)** | **1.5.0 (colorManagement: true)** |
|:------------------------------:|:--------------------------------------:|:-------------------------------------:|:----------------------------------:|:---------------------------------:|
| `new THREE.Color(...)`         |              ✅                   | ❌                             | ✅                            | ✅                           |
| `.applyColorCorrection(color)` | ✅                                | ✅                               | ✅                            | ✅                           |
| `color.convertSRGBToLinear()`  | ❌                              | ✅                               | ❌                          | ❌                         |

Of course, this assumes that the (input) colour is intended to be sRGB, but components might very well be constructing Colors using hex notation, but expecting it to be linear or, like aframe-environment-component does, blend sRGB colours. 🫤

So it can't be entirely seamless, but we should document the changes properly. Not sure what the common approach for backwards/forwards compatibility and breaking changes is for A-Frame, but it would be nice for it to be possible to make components compatible with both 1.4.x and 1.5.0 in terms of colours.

Reply:
Oh I see `applyColorCorrection` has a condition on the conversion, I'll try to modify my components to use that instead of `color.convertSRGBToLinear()` myself to be compatible with both 1.4.0 and 1.5.0. Thanks.